server/src/child.rs

```diff
@@ -15,11 +15,14 @@ pub struct Thread {
     pub thread: JoinHandle<()>,
     pub tx: SyncSender<ServerMessage>,
     pub rx: Receiver<DebugMessage>,
-
+   
     pub execution: Option<i32>,
     pub id: i32,
+    pub e_type: String,
```

Comment:
Should be an enum rather than a `String`; should also merge with the `execution` field above.